apps/studio/src/utils/sitemap.ts

```diff
@@ -0,0 +1,101 @@
+import type { IsomerSitemap } from "@opengovsg/isomer-components"
+import type { Resource } from "@prisma/client"
+
+type ResourceDto = Omit<
+  Resource,
+  "id" | "parentId" | "publishedVersionId" | "draftBlobId"
+> & {
+  id: string
+  parentId: string | null
+}
+
+const getSitemapTreeFromArray = (
+  resources: ResourceDto[],
+  parentId: string | null,
+  path: string,
+): IsomerSitemap[] | undefined => {
+  if (resources.length === 0) {
+    return undefined
+  }
+
+  const children = resources.filter((resource) => {
+    if (parentId === null) {
+      return resource.parentId === null && resource.permalink !== ""
+    }
+    return resource.parentId === parentId
+  })
+
+  // Filter out children that have both the Page and the Folder types and keep
+  // only the one with the Folder type, then reassign the ID to the ID of the
+  // resource with the Page type
+  return children
+    .filter(
+      (child) =>
+        !(
+          child.type === "Page" &&
+          children.some(
+            (otherChild) =>
+              otherChild.id !== child.id &&
+              otherChild.permalink === child.permalink &&
+              otherChild.type === "Folder",
+          )
+        ),
+    )
+    .map((resource) => {
+      const permalink = `${path}${resource.permalink}`
+      if (resource.type === "Folder") {
+        const idOfPage = children.find(
+          (child) =>
+            child.id !== resource.id &&
+            child.permalink === resource.permalink &&
+            child.type === "Page",
+        )?.id
```

Comment:
abit confused here, why is the method used `.find`, which suggests that only the first page matters? also, why not just use `child.parentId === resource.id`?

Reply:
not entirely sure here, but reading at first glance seems to suggest we're trying to find another page iwth the same permalink as the current folder but the earlier filter seems to omit pages that have some folder iwth the same permalink, so not entirely sure why this works

Reply:
Yup, because there should only be 1 result. The .find is done on the original array, so it still contains the entries with the actual Page.